CRM/Member/BAO/MembershipType.php

```diff
@@ -91,11 +91,22 @@ public static function setIsActive($id, $is_active) {
    * @param array $ids
    *   Array contains the id (deprecated).
    *
-   *
-   * @return object
+   * @return \CRM_Member_DAO_MembershipType
+   * @throws \CiviCRM_API3_Exception
    */
   public static function add(&$params, $ids = array()) {
-    $id = CRM_Utils_Array::value('id', $params, CRM_Utils_Array::value('membershipType', $ids));
+    if (empty($params['id'])) {
+      $params['id'] = CRM_Utils_Array::value('membershipType', $ids);
```

Comment:
should we add a deprecation notice here?

Reply:
Yes, added.